cachi2/core/package_managers/pip.py

```diff
@@ -103,6 +110,39 @@ def fetch_pip_source(request: Request) -> RequestOutput:
     )
 
 
+def _generate_purl(package: dict[str, Any], package_path: RootedPath = None) -> str:
+    """Get the purl for this package."""
+    type = "pypi"
+    name = package["name"]
+    version = None
+    subpath = None
+    if package_path:
+        url = get_repo_id(package_path.root).as_vcs_url_qualifier()
+        qualifiers = {"vcs_url": url}
+        if package_path.subpath_from_root != Path("."):
+            subpath = f"subdirectory={package_path.subpath_from_root}"
+    if package.get("kind", None) == "pypi":
```

Comment:
I would prefer separate logic for the main package vs. dependencies

`package_path` is only used when processing the main package
the `package` dict only has "kind" when processing dependencies

It also wouldn't hurt to include the `version` of the main package in the main package purl